pkgs/development/python-modules/python-imports-check-libstdcxx/test_forward_compatibility_fails.py

```diff
@@ -0,0 +1,39 @@
+import os
+import sys
+
+assert (
+    "EXPECTED_LIBSTDCXX" in os.environ
+), "The derivation failed to export EXPECTED_LIBSTDCXX"
+
+expected_libstdcxx_root = os.environ["EXPECTED_LIBSTDCXX"]
+
+import load_libstdcxx_old
+import psutil
+
+try:
+    import load_libstdcxx_new
+except ImportError:
+    print(f"python-imports-check-load-libstdcxx successfuly fails", file=sys.stderr)
+    sys.exit(0)
+print(
+    "python-imports-check-load-libstdcxx doesn't fail the naive forward"
+    " compatibility test, it probably needs to be updated.",
+    file=sys.stderr,
+)
+
+loaded_libraries = [x.path for x in psutil.Process(os.getpid()).memory_maps()]
+loaded_libraries = [x for x in loaded_libraries if "libstdc++" in x]
+
+assert loaded_libraries, "Didn't load either version of libstdc++"
+
+assert all(x.startswith(expected_libstdcxx_root) for x in loaded_libraries), (
+    "The python interpreter didn't load the old version of libstdc++:"
+    f" {loaded_libraries=} expected one from {expected_libstdcxx_root}"
+)
+
+print(
+    "The python interpreter has correctly loaded the old version of"
+    f" libstdc++ as expected: {loaded_libraries}",
+    file=sys.stderr,
+)
+sys.exit(1)
```

Comment:
The state strikes back again. Yup, definitely refactoring this.

Here's an explanation before I myself have forgot it.

On line 18 we have already printed the error message that the test appears to be infallible (=useless).

Then, the reason we expected to see an `ImportError` is because we have forced python to load the old libstdc++ in the beginning of the program, and then tried to import `load_libstdcxx_new` that couldn't look up the symbols it needed.

Now what does it mean if we have reached this last print? It means that our tricks worked, and we have indeed loaded the old libstdc++ first, but our `load_libstdcxx_new` didn't require any new symbols.

This is different from the errors the asserts above signal